### DISCUSSION TITLE
Randomly removing MMR for no reason

### DISCUSSION THREAD

**Discussion by Onisseur** [Heroes-Profile/heroesprofile#1045] [General]
Is this happening to anyone else? The website seemingly removes MMR even though I win games and it happens quite regularly, but it also some times adds extra MMR
![bbfb2431ec4eab1f5a028c41c612f57d](https://github.com/user-attachments/assets/22e21468-3d98-4172-a1d1-aa92cab47de5)

**Comment by Zemill** [comment]
Appears you are uploading your replays out of order.  Are you using the web uploader and choosing which games to upload?

We suggest using the automatic uploader, you wouldnt have this issue

Check your MMR on the MMR page, see if it looks more consistent

**Reply by Onisseur** [reply]
I upload my games as I finish playing them and I use the web uploader, yes. I do skip my losses after this MMR thing happens since it feels quite unfair to loose it out of nowhere. However, it also happens if I do upload all of my games without skipping any. 
![a622bdf15ef090e49d2d35528e586b5f](https://github.com/user-attachments/assets/7629f861-8ab7-4477-bedc-4914261a3f40)
I will start using the automatic uploader, thanks for the suggestion.

**Reply by Zemill** [reply]
provide me your account and I can take a look

**Reply by Onisseur** [reply]
https://www.heroesprofile.com/Player/Onisseur/11938874/2

**Reply by Zemill** [reply]
Alright, so the cause of this is as expected, replays being uploaded out of order.  Your MMR is correct.

So lets use the Valla game as our starting point

![image](https://github.com/user-attachments/assets/cad7259b-413b-4e7d-bd20-f0609dc9008a)

In the image you can see the Valla game MMR circled, and if you follow the MMR up the table then you are not losing or gaining MMR in unexpected ways.  The one thing you can notice is the second column, which is game date.  See how it is not consistent and not in chronological order? If you look at the third column you can see that you (or someone else) uploaded your games in that non-chronological order.  MMR is calculated on first in, first out, and that process has no knowledge that you are uploading stuff out of order.

The discrepancy you see on the profile page is because the profile page is ordered on game date, and this Valla game is a newer game uploaded way before all of those other games that are older than it.

Here is an image showing it

![image](https://github.com/user-attachments/assets/181ed26f-1287-43b3-9459-2fb609202ff8)

See that there are a ton of games between the Valla game and the Stukov game?   All those games between Valla and Stukov were uploaded out of order.  They should have been uploaded before the Valla game.


So, all of this to say, using the automatic uploader will resolve this, otherwise if you use the manual one, make sure to upload in order and all of your games.  **You actually inadvertently did this to yourself by selectively choosing what games to upload or not, since others could decide to upload those games for you.**

Note:  I usually ban people for selectively choosing what games to upload as they generally do it to game the leaderboards.   Based on our conversation so far, it appears you were not doing it maliciously, so ill give you a pass this time, just make sure to make your best attempts to upload all of the replays.

**Reply by Zemill** [reply]
I forgot to mention.  It is a valid condition to lose MMR on a win if you/your team drastically out rank your opponent.  You can see this occurring to you in the Malthael game where you lost .20 MMR on a win

**Comment by Onisseur** [comment]
Thank you so much for looking through my problem and explaining it Zemill. It has been bothering me a lot and I have already downloaded the automatic uploader and started using it.
